target/linux/realtek/image/rtl838x.mk

```diff
@@ -411,6 +411,15 @@ define Device/zyxel_gs1900-24-v1
 endef
 TARGET_DEVICES += zyxel_gs1900-24-v1
 
+define Device/zyxel_gs1900-24-v2
+  $(Device/zyxel_gs1900)
+  SOC := rtl8382
+  DEVICE_MODEL := GS1900-24
+  DEVICE_VARIANT := v2
+  ZYXEL_VERS := AAHL
```

Comment:
Without `SUPPORTED_DEVICES += zyxel,gs1900-24-v1` users will have to force upgrade existing installs on B1 hardware. The system configuration could be restored, but you may want to mention this in the commit message.

To be clear, I wouldn't add `SUPPORTED_DEVICES`, as it (hopefully) makes the user think about what they are about to do.